File: Models/DSSAT_EPICST_standalone/src/cs/DSSAT_EPICST_standalone/STEMP_EPIC_State.cs
using System;
using System.Collections.Generic;
public class STEMP_EPIC_State 
{
    private double[] _ST = new double[NL];
    private double[] _TMA = new double[5];
    private double _SRFTEMP;
    private int _NDays;
    private double[] _DSMID = new double[NL];
    private double _CUMDPT;
    private double _X2_PREV;
    private double _TDL;
    private int[] _WetDay = new int[30];
    
        public STEMP_EPIC_State() { }
    
    
    public STEMP_EPIC_State(STEMP_EPIC_State toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    ST = new double[NL];
            for (int i = 0; i < NL; i++)
            { ST[i] = toCopy.ST[i]; }
    
    TMA = new double[5];
            for (int i = 0; i < 5; i++)
            { TMA[i] = toCopy.TMA[i]; }
    
    SRFTEMP = toCopy.SRFTEMP;
    NDays = toCopy.NDays;
    DSMID = new double[NL];
            for (int i = 0; i < NL; i++)
            { DSMID[i] = toCopy.DSMID[i]; }
    
    CUMDPT = toCopy.CUMDPT;
    X2_PREV = toCopy.X2_PREV;
    TDL = toCopy.TDL;
    WetDay = new int[30];
            for (int i = 0; i < 30; i++)
            { WetDay[i] = toCopy.WetDay[i]; }
    
    }
    }
    public double[] ST
        {
            get { return this._ST; }
            set { this._ST= value; } 
        }
    public double[] TMA
        {
            get { return this._TMA; }
            set { this._TMA= value; } 
        }
    public double SRFTEMP
        {
            get { return this._SRFTEMP; }
            set { this._SRFTEMP= value; } 
        }
    public int NDays
        {
            get { return this._NDays; }
            set { this._NDays= value; } 
        }
    public double[] DSMID
        {
            get { return this._DSMID; }
            set { this._DSMID= value; } 
        }
    public double CUMDPT
        {
            get { return this._CUMDPT; }
            set { this._CUMDPT= value; } 
        }
    public double X2_PREV
        {
            get { return this._X2_PREV; }
            set { this._X2_PREV= value; } 
        }
    public double TDL
        {
            get { return this._TDL; }
            set { this._TDL= value; } 
        }
    public int[] WetDay
        {
            get { return this._WetDay; }
            set { this._WetDay= value; } 
        }
}

File: Models/DSSAT_EPICST_standalone/src/cpp1/DSSAT_EPICST_standalone/STEMP_EPIC_State.cpp
#include "STEMP_EPIC_State.h"
STEMP_EPIC_State::STEMP_EPIC_State() { }

vector<double> & STEMP_EPIC_State::getST() {return this-> ST; }
vector<double> & STEMP_EPIC_State::getTMA() {return this-> TMA; }
double STEMP_EPIC_State::getSRFTEMP() {return this-> SRFTEMP; }
int STEMP_EPIC_State::getNDays() {return this-> NDays; }
vector<double> & STEMP_EPIC_State::getDSMID() {return this-> DSMID; }
double STEMP_EPIC_State::getCUMDPT() {return this-> CUMDPT; }
double STEMP_EPIC_State::getX2_PREV() {return this-> X2_PREV; }
double STEMP_EPIC_State::getTDL() {return this-> TDL; }
vector<int> & STEMP_EPIC_State::getWetDay() {return this-> WetDay; }

void STEMP_EPIC_State::setST(vector<double> const & _ST){
    this->ST = _ST;
}
void STEMP_EPIC_State::setTMA(vector<double> const & _TMA){
    this->TMA = _TMA;
}
void STEMP_EPIC_State::setSRFTEMP(double _SRFTEMP) { this->SRFTEMP = _SRFTEMP; }
void STEMP_EPIC_State::setNDays(int _NDays) { this->NDays = _NDays; }
void STEMP_EPIC_State::setDSMID(vector<double> const & _DSMID){
    this->DSMID = _DSMID;
}
void STEMP_EPIC_State::setCUMDPT(double _CUMDPT) { this->CUMDPT = _CUMDPT; }
void STEMP_EPIC_State::setX2_PREV(double _X2_PREV) { this->X2_PREV = _X2_PREV; }
void STEMP_EPIC_State::setTDL(double _TDL) { this->TDL = _TDL; }
void STEMP_EPIC_State::setWetDay(vector<int> const & _WetDay){
    this->WetDay = _WetDay;
}